src/main/java/core/basesyntax/FruitShopApp.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import core.basesyntax.service.converter.DataConverter;
+import core.basesyntax.service.converter.DataConverterImpl;
+import core.basesyntax.service.handler.*;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.report.ReportGenerator;
+import core.basesyntax.service.report.ReportGeneratorImpl;
+import core.basesyntax.service.shop.ShopService;
+import core.basesyntax.service.shop.ShopServiceImpl;
+import core.basesyntax.service.strategy.OperationStrategy;
+import core.basesyntax.service.strategy.OperationStrategyImpl;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.util.filereader.FileReader;
+import core.basesyntax.util.filereader.FileReaderImpl;
+import core.basesyntax.util.filewriter.FileWriter;
+import core.basesyntax.util.filewriter.FileWriterImpl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopApp {
+    public static void main(String[] arg) {
+        // 1. Read the data from the input CSV file
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.read("reportToRead.csv");
+
+        // 2. Convert the incoming data into FruitTransactions list
+        DataConverter dataConverter = new DataConverterImpl();
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        // 3. Create and feel the map with all OperationHandler implementations
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+
+        // 4. Process the incoming transactions with applicable OperationHandler implementations
+        Storage storage = new Storage();
+        ShopService shopService = new ShopServiceImpl(operationStrategy, storage);
+        shopService.process(transactions);
+
+        // 5.Generate report based on the current Storage state
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
+        String resultingReport = reportGenerator.getReport(storage.getProducts());
+
+        // 6. Write the received report into the destination file
```

Comment:
Don't begin class or method implementation with an empty line. Please remove this redundant empty line.